clang/test/CheckedCRewriter/arrboth.c

```diff
@@ -107,54 +89,49 @@ int zerohuh(int n) {
 }
 
 int *mul2(int *x) { 
+	//CHECK: _Ptr<int> mul2(_Ptr<int> x) { 
     *x *= 2; 
     return x;
 }
 
-//CHECK_NOALL: _Ptr<int> mul2(_Ptr<int> x) { 
-
-//CHECK_ALL: _Ptr<int> mul2(_Ptr<int> x) { 
-
 int * sus(int * x, int * y) {
+	//CHECK_NOALL: int * sus(int *x, _Ptr<int> y) {
+	//CHECK_ALL: _Nt_array_ptr<int> sus(int *x, _Ptr<int> y) {
 x = (int *) 5;
+	//CHECK: x = (int *) 5;
         int *z = calloc(5, sizeof(int)); 
-        int i, *p, fac;
+	//CHECK_NOALL: int *z = calloc<int>(5, sizeof(int)); 
+	//CHECK_ALL: _Nt_array_ptr<int> z : count(5) =  calloc<int>(5, sizeof(int)); 
+        int i, fac;
+        int *p;
+	//CHECK_NOALL: int *p;
+	//CHECK_ALL: _Array_ptr<int> p : count(5) = ((void *)0);
         for(i = 0, p = z, fac = 1; i < 5; ++i, p++, fac *= i) 
         { *p = fac; }
 z += 2;
 return z; }
-//CHECK_NOALL: int * sus(int *x, _Ptr<int> y) {
-//CHECK_NOALL:         int *z = calloc<int>(5, sizeof(int)); 
-//CHECK_NOALL:         int i, *p, fac;
-//CHECK_ALL: _Nt_array_ptr<int> sus(int *x, _Ptr<int> y) {
-//CHECK_ALL:         _Nt_array_ptr<int> z : count(5) =  calloc<int>(5, sizeof(int)); 
-//CHECK_ALL: _Array_ptr<int> p : count(5) = ((void *)0);
 
 int * foo() {
+	//CHECK_NOALL: int * foo(void) {
+	//CHECK_ALL: _Nt_array_ptr<int> foo(void) {
         int * x = malloc(sizeof(int));
+	//CHECK: int * x = malloc<int>(sizeof(int));
```

Comment:
This is probably not an issue with your test, but: It seems weird to me that we'd add a type instantiation for `malloc` if we were returning a non-Checked type. Maybe this is an issue that @jackastner should take a look at?

Reply:
Is the thinking here that if we're not going to convert it, we should leave the source code unchanged? This isn't wrong, is it?

Reply:
Yes, that's the thinking. I don't think it's wrong --- if it was, your tests wouldn't have compiled, right?